File: 0274. H-Index/Solution0274.py
class Solution:
    def hIndex0(self, citations: List[int]) -> int:
        citations.sort()
        res, idx = 0, len(citations)-1
        while idx >= 0 and citations[idx] > res:
            res+=1
            idx-=1
        return res  

    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        count = [0]* (n+1)
        for c in citations:
            count[min(c,n)]+=1
                
        tot = 0
        for i in range(n,-1,-1):
            tot += count[i]
            if tot >= i: #definition
                return i
        return 0

File: 0274. H-Index/Solution0274.java
class Solution274 {
    public int hIndex(int[] citations) {
        int n = citations.length;
        int[] count = new int[n+1];
        for (int i = 0; i < n; i++) {
            if (citations[i] >= n) {
                count[n]++;
            } else {
                count[citations[i]]++;
            }
        }
        int tot = 0;
        for (int i = n; i >= 0; i--) {
            tot += count[i];
            if (tot >= i) {
                return i;
            }
        }
        return 0;
    }
    
    public int hIndex0(int[] citations) {
        Arrays.sort(citations);
        int i = citations.length-1, res = 0;
        while (i>=0 && citations[i] > res) {
            res++; i--;
        }
        return res;
    }
}